src/Modal.js

```diff
@@ -226,28 +193,12 @@ const ModalMarkup = React.createClass({
   },
 
   renderContent() {
-    let children = toChildArray(this.props.children); // b/c createFragment is in addons and children can be a key'd object
-    let hasNewHeader = children.some( c => c.type.__isModalHeader);
-
-    if (!hasNewHeader && this.props.title != null){
-      deprecationWarning(
-        'Specifying `closeButton` or `title` Modal props',
-        'the new Modal.Header, and Modal.Title components');
-
-      children.unshift(
-        <Header closeButton={this.props.closeButton} onHide={this._getHide()}>
-          { this.props.title &&
-            <Title>{this.props.title}</Title>
-          }
-        </Header>
-      );
-    }
 
-    return React.Children.map(children, child => {
+    return React.Children.map(this.props.children, child => {
```

Comment:
why don't you use `mapValidComponents()`?
[src/utils/ValidComponentChildren.js#L15](https://github.com/react-bootstrap/react-bootstrap/blob/02a8764b196b9268678ded26838a2436a54a2090/src/utils/ValidComponentChildren.js#L15)